askama_derive/src/input.rs

```diff
@@ -25,106 +37,36 @@ impl<'a> TemplateInput<'a> {
-        // understand. Raise panics if something is not right.
-        // `source` contains an enum that can represent `path` or `source`.
-        let mut source = None;
-        let mut print = Print::None;
-        let mut escaping = None;
-        let mut ext = None;
-        let mut syntax = None;
-        for item in meta_list.nested {
-            let pair = match item {
-                syn::NestedMeta::Meta(syn::Meta::NameValue(ref pair)) => pair,
-                _ => panic!(
-                    "unsupported attribute argument {:?}",
-                    item.to_token_stream()
-                ),
-            };
-
-            if pair.path.is_ident("path") {
-                if let syn::Lit::Str(ref s) = pair.lit {
-                    if source.is_some() {
-                        panic!("must specify 'source' or 'path', not both");
-                    }
-                    source = Some(Source::Path(s.value()));
-                } else {
-                    panic!("template path must be string literal");
-                }
-            } else if pair.path.is_ident("source") {
-                if let syn::Lit::Str(ref s) = pair.lit {
-                    if source.is_some() {
-                        panic!("must specify 'source' or 'path', not both");
-                    }
-                    source = Some(Source::Source(s.value()));
-                } else {
-                    panic!("template source must be string literal");
-                }
-            } else if pair.path.is_ident("print") {
-                if let syn::Lit::Str(ref s) = pair.lit {
-                    print = s.value().into();
-                } else {
-                    panic!("print value must be string literal");
-                }
-            } else if pair.path.is_ident("escape") {
-                if let syn::Lit::Str(ref s) = pair.lit {
-                    escaping = Some(s.value());
-                } else {
-                    panic!("escape value must be string literal");
-                }
-            } else if pair.path.is_ident("ext") {
-                if let syn::Lit::Str(ref s) = pair.lit {
-                    ext = Some(s.value());
-                } else {
-                    panic!("ext value must be string literal");
-                }
-            } else if pair.path.is_ident("syntax") {
-                if let syn::Lit::Str(ref s) = pair.lit {
-                    syntax = Some(s.value())
-                } else {
-                    panic!("syntax value must be string literal");
-                }
-            } else {
-                panic!(
-                    "unsupported attribute key '{}' found",
-                    pair.path.to_token_stream()
-                )
-            }
-        }
+        assert_ne!(
+            source.is_some(),
+            path.is_some(),
+            "One of path or source must exist and they are mutually exclusive",
```

Comment:
Let's lowercase the `One` here to be consistent, and simplify the "and they are mutually exclusive to " (but not both)"?